autoscale/src/test/java/com/sequenceiq/periscope/InjectTest.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.periscope;
+
+import static java.lang.System.lineSeparator;
+
+import java.lang.reflect.Member;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.reflections.Reflections;
+import org.reflections.scanners.FieldAnnotationsScanner;
+import org.reflections.scanners.MemberUsageScanner;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.scanners.TypeAnnotationsScanner;
+
+public class InjectTest {
```

Comment:
General note: The whole logic seems to be copy-pasted in every app. Would it be perhaps possible to extract the majority into some shared lib (e.g. `common:test`), so that these app-specific tests could become significantly shortened? A similar approach was followed recently with the various `FlowPayloadSerializabilityTest` app tests that all rely on the shared logic of `FlowPayloadSerializabilityChecker` in `flow:test`.

Reply:
sure